compiler/rustc_mir_transform/src/inline.rs

```diff
@@ -39,11 +39,7 @@ struct CallSite<'tcx> {
 
 /// Returns true if MIR inlining is enabled in the current compilation session.
 crate fn is_enabled(tcx: TyCtxt<'_>) -> bool {
-    if let Some(enabled) = tcx.sess.opts.debugging_opts.inline_mir {
-        return enabled;
-    }
-
-    tcx.sess.mir_opt_level() >= 3
+    tcx.sess.opts.debugging_opts.inline_mir && tcx.sess.mir_opt_level() >= 3
```

Comment:
This changes the condition. Before it would always run if `-Zinline-mir=yes` is used, never if `-Zinline-mir=no` is used and only for mir opt level >= 3 if `-Zinline-mir` is not specified. After it will only run when `-Zinline-mir=yes` and mir opt level >= 3 is combined.